firefox_puppeteer/ui/toolbars.py

```diff
@@ -38,13 +42,18 @@ def home_button(self):
         """
         return self.marionette.find_element(By.ID, 'home-button')
 
-    @use_class_as_property('ui.toolbars.LocationBar')
+    @property
     def locationbar(self):
         """Provides access to the DOM elements contained in the
         locationbar.
 
         See the :class:`LocationBar` reference.
         """
+        if not self._locationbar:
+            from .toolbars import LocationBar
+            self._locationbar = LocationBar(lambda: self.marionette, self.window, self.element)
```

Comment:
When you do this the urlbar property can return `self.element` given that we cache that.